### PR TITLE
remove, removeOne and splice methods & tests

### PR DESCRIPTION
**Pull request by noevents**

### CODE REVIEW

**Comment by Salakar** [comment]
Would you mind sorting the linting issues if possible please, the ones listed under `index.js` header here: https://www.codacy.com/app/mike-diarmid/denque/pullRequest?prid=737043 - You can ignore the `complexity` ones (unless you can get around them).

**Comment by noevents** [comment]
@Salakar Can you merge pr now?

File: index.js [hunk 1]
```diff
@@ -162,6 +162,193 @@ Denque.prototype.pop = function pop() {
...
+Denque.prototype.splice = function splice(index, count) {
+  var i = index;
+  var size = this.size();
+  // expect a number or return undefined
+  if ((i !== (i | 0))) {
+    return void 0;
+  }
+  if (this._head === this._tail) return void 0;
+  if (i > size || i < -size) return void 0;
+  if (i === size && count != 0) return void 0;
```
Comment (Salakar) [inline index.js]: Not entirely sure what this line is trying to do? Could you clarify - your PR is missing quite a bit of coverage so i'm trying to bring it back up to 99%~

Comment (Salakar) [inline index.js]: Can't release until then

File: index.js [hunk 2]
```diff
@@ -162,6 +162,193 @@ Denque.prototype.pop = function pop() {
...
+    } else {
+      temp = new Array(size - (i + count));
+      var leng = temp.length;
+      for(k = 0; k < leng; k++){
+        temp[k] = this._list[(this._head + i + count + k) & this._capacityMask];
+      }
+      if(count === 0){
+        removed = [];
+        if(i != size){
+          this._tail = (i - 1 + len) & this._capacityMask;
```
Comment (Salakar) [inline index.js]: This line is also missing coverage

**Comment by Salakar** [comment]
I've pushed up a few changes to your PR, if you could finish the rest from my review above that'd be great, thanks

**Comment by noevents** [comment]
Thanks for review!
I've fixed coverage and piece of code from this review: https://github.com/invertase/denque/pull/7#pullrequestreview-52394569

Without line from here: https://github.com/invertase/denque/pull/7#pullrequestreview-52392317
node will throw RangeError

**Pull request merged**